assets/app/scripts/directives/oscFileInput.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .directive('oscFileInput', function(Logger) {
+    return {
+      restrict: 'E',
+      scope: {
+        name: "@",
+        model: "=",
+        required: "="
+      },
+      templateUrl: 'views/directives/osc-file-input.html',
+      link: function(scope, element, attrs){
+        scope.supportsFileUpload = (window.File && window.FileReader && window.FileList && window.Blob);
+        $(element).change(function(){
+          var file = $('input[type=file]',this)[0].files[0];
+          scope.fileName = file.name;
+          scope.$apply(function(){
+            var reader = new FileReader();
+            reader.onloadend = function(){
+              scope.model = reader.result;
+            };
+            reader.onerror = function(e){
+              Logger.error(e);
```

Comment:
not sure what to do here... we can't just leave the file selected and not tell them about the error... maybe flip `scope.supportsFileUpload` to false so they can paste in the file? If we do that, we should also notify them that there was an error reading the file or they'll be really confused